pkg/controllers/override/imageparse.go

```diff
@@ -0,0 +1,176 @@
+	var newTag string
+	switch operator {
+	case OperatorAdd, OperatorReplace:
+		newTag = value
+	case OperatorRemove:
+		newTag = ""
+	}
+
+	if operator != OperatorRemove && !utilreference.AnchoredTagRegexp.MatchString(newTag) {
+		return fmt.Errorf("invalid tag format after applying the overrider")
+	}
+
+	i.tag = newTag
+	return nil
+}
+
+func (i *Image) OperateDigest(operator, value string) error {
+	var newDigest string
+	switch operator {
+	case OperatorAdd, OperatorReplace:
+		newDigest = value
+	case OperatorRemove:
+		newDigest = ""
+	}
+
+	if operator != OperatorRemove && !utilreference.AnchoredDigestRegexp.MatchString(newDigest) {
+		return fmt.Errorf("invalid digest format after applying the overrider")
+	}
+
+	i.digest = newDigest
+	return nil
+}
+
+func (i *Image) String() string {
+	fullName := i.repository
+
+	if i.registry != "" {
+		fullName = i.registry + "/" + i.repository
+	}
+
+	if i.tag != "" {
+		fullName = fullName + ":" + i.tag
+	}
+
+	if i.digest != "" {
+		fullName = fullName + "@" + i.digest
+	}
+
+	return fullName
+}
+
+// ParseImage returns an Image from the given string.
+func ParseImage(fullName string) (*Image, error) {
+	ref, err := reference.Parse(fullName)
+	if err != nil {
+		return nil, err
+	}
+
+	image := &Image{}
+	if named, ok := ref.(reference.Named); ok {
+		image.registry, image.repository = SplitRegistryAndRepository(named.Name())
+	}
+
+	if tagged, ok := ref.(reference.Tagged); ok {
+		image.tag = tagged.Tag()
+	}
+
+	if digested, ok := ref.(reference.Digested); ok {
+		image.digest = digested.Digest().String()
+	}
+
+	return image, nil
+}
+
+// SplitRegistryAndRepository refers to https://github.com/distribution/distribution/blob/v2.7.1/reference/normalize.go#L62
+func SplitRegistryAndRepository(name string) (registry, repository string) {
+	i := strings.IndexRune(name, '/')
+	if i == -1 || (!strings.ContainsAny(name[:i], ".:") && name[:i] != "localhost") {
+		registry, repository = "", name
+	} else {
+		registry, repository = name[:i], name[i+1:]
+	}
+	return
+}
+
+func GetStringFromUnstructuredObj(input *unstructured.Unstructured, path string) (string, error) {
+	fields := strings.Split(strings.Trim(path, pathSeparator), pathSeparator)
+	var currentObj interface{} = input.Object
+
+	for i, field := range fields {
+		switch obj := currentObj.(type) {
+		case map[string]interface{}:
+			currentObj = obj[field]
+		case []interface{}:
+			index, err := strconv.ParseInt(field, 10, 32)
+			if err != nil {
+				return "", fmt.Errorf(
+					"slice found at path %s, but the next path segment is not a number (found %q)",
+					strings.Join(fields[:i], "/"),
+					field,
+				)
+			}
+			if index < 0 || int(index) >= len(obj) {
+				return "", fmt.Errorf("index(%d) is negative or out of range", index)
+			}
+			currentObj = obj[index]
+		default:
+			return "", fmt.Errorf(
+				"%s accessor error: unkowen type %T, expected map[string]interface{} or []interface{}",
+				strings.Join(fields[:i+1], "/"),
```

Comment:
```suggestion
				strings.Join(fields[:i], "/"),
```
Are we off by 1 here?

Reply:
yes, this should be `i`, the code has been updated.